spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/ConverterAwareMappingSpannerEntityWriter.java

```diff
@@ -141,11 +140,20 @@ public void write(Object source, MultipleValueBinder sink,
 				.getPropertyAccessor(source);
 		persistentEntity.doWithProperties(
 				(PropertyHandler<SpannerPersistentProperty>) spannerPersistentProperty -> {
-					if (!writeAllColumns && !includeColumns
-							.contains(spannerPersistentProperty.getColumnName())) {
-						return;
+					if (spannerPersistentProperty.isEmbedded()) {
+						Object embeddedObject = accessor
+								.getProperty(spannerPersistentProperty);
+						if (embeddedObject != null) {
+							write(embeddedObject, sink, includeColumns);
+						}
+					}
+					else {
+						if (!writeAllColumns && !includeColumns
```

Comment:
Please refactor this block to make positive checks instead of negative, to improve readability. This construct
```
if (!a && !b){
    return;
}
code 
```
can be expressed in this way:
```
if (a || b){
    code
} 
```
Also, after this modification it would probably makes sense to merge that `if` with previous `else`.

Reply:
will fix!